go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -118,6 +123,84 @@ func TestComputeFilesWithXattr(t *testing.T) {
 	}
 }
 
+func TestComputeFileDigestWithXattr(t *testing.T) {
+	xattrDgName := "user.myhash"
+	overwriteXattrDgName(t, xattrDgName)
+	tests := []struct {
+		filename   string
+		contents   string
+		xattrDgStr string
+		wantDgStr  string
+		wantErr    bool
+	}{
+		{
+			filename:  "provide_no_xattr_will_generate_real_digest(sha256sum+real_size)",
+			contents:  "123456",
+			wantDgStr: "8d969eef6ecad3c29a3a629280e686cf0c3f5d5a86aff3ca12020c923adc6c92/6",
+		},
+		{
+			filename:   "provide_only_digest_hash_will_generate_digest_with_real_size",
+			contents:   "123456",
+			xattrDgStr: "1111111111111111111111111111111111111111111111111111111111111111",
+			wantDgStr:  "1111111111111111111111111111111111111111111111111111111111111111/6",
+		},
+		{
+			filename:   "valid_full_digest_(hash+size)_in_xatrr_will_be_used_directly",
+			contents:   "",
+			xattrDgStr: "1111111111111111111111111111111111111111111111111111111111111111/666",
+			wantDgStr:  "1111111111111111111111111111111111111111111111111111111111111111/666",
+		},
+		{
+			filename:   "provide_invalid_digest_(hash_missing_digits)_will_set_md.Err",
+			contents:   "123456",
+			xattrDgStr: "abc",
+			wantDgStr:  "abc/6",
+			wantErr:    true,
+		},
+		{
+			filename:   "provide_invalid_full_digest_(hash_missing_digits)_will_set_md.Err",
+			contents:   "123456",
+			xattrDgStr: "666/666",
+			wantDgStr:  "666/666",
+			wantErr:    true,
+		},
+		{
+			filename:   "provide_invalid_full_digest_(extra_slash)_will_set_md.Err",
+			contents:   "123456",
+			xattrDgStr: "///666",
+			wantDgStr:  "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/0",
+			wantErr:    true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.filename, func(t *testing.T) {
+			path := tc.filename
+			//In the Github pipeline, set xatrr for files created under /tmp folder (by running t.tempDir()) will raise an error for operation not supported
+			//But local run of the unit test will pass. So we have to create these test files under the PWD.
+			err := os.WriteFile(path, []byte(tc.contents), 0666)
+			if err != nil {
+				t.Fatalf("Failed to write to file: %v\n", err)
+			}
+			defer os.RemoveAll(path)
```

Comment:
t.Cleanup is the recommended pattern for tests.

Reply:
I am using t.Cleanup() to delete test files now. Also changed this for all the other unit tests inside of this file for consistency.